### ISSUE TITLE
Add scrub pause

### ISSUE THREAD

**Issue by kdave** [kdave/btrfs-progs#234]
Add `scrub pause` as a more logical and convenient counterpart for `resume`. This is similar to what balance command interface has. The scrub pause is effectively the same thing as current cancel that saves the last state.

The sequence of `start` -> `pause` -> `resume` is more user friendly.

Current implementation of `scrub cancel` must stay unchanged as it is widely used.

**Comment by danobi** [comment]
I briefly looked around the code -- this will require kernel changes, right?

It seems we'd need a new BTRFS_IOC_SCRUB_PAUSE ioctl to save state to disk just like BTRFS_BALANCE_CTL_PAUSE, right?

**Comment by kdave** [comment]
No, the scrub state is saved externally in /var/lib/btrfs/scrub.status.UUID
